src/approaches/analysis_approach.py

```diff
@@ -0,0 +1,274 @@
+        # Python objects during parsing.
+        option_name_to_option = {op.name: op for op in self._initial_options}
+        obj_name_to_obj = {o.name: o for o in objects}
+        # We assume the LLM's output is such that each line contains a
+        # option_name(obj0:type0, obj1:type1,...).
+        options_str_list = llm_prediction.split('\n')
+        for option_str in options_str_list:
+            option_str_stripped = option_str.strip()
+            option_name = option_str_stripped.split('(')[0]
+            # Skip empty option strs.
+            if not option_str:
+                continue
+            if option_name not in option_name_to_option.keys():
+                logging.info(
+                    f"Line {option_str} output by LLM doesn't "
+                    "contain a valid option name. Terminating option plan "
+                    "parsing.")
+                break
+            option = option_name_to_option[option_name]
+            # Now that we have the option, we need to parse out the objects
+            # along with specified types.
+            typed_objects_str_list = option_str_stripped.split('(')[1].strip(
+                ')').split(',')
+            objs_list = []
+            malformed = False
+            for i, type_object_string in enumerate(typed_objects_str_list):
+                object_type_str_list = type_object_string.strip().split(':')
+                # We expect this list to be [object_name, type_name].
+                if len(object_type_str_list) != 2:
+                    logging.info(f"Line {option_str} output by LLM has a "
+                                 "malformed object-type list.")
+                    malformed = True
+                    break
+                object_name = object_type_str_list[0]
+                type_name = object_type_str_list[1]
+                if object_name not in obj_name_to_obj.keys():
+                    logging.info(f"Line {option_str} output by LLM has an "
+                                 "invalid object name.")
+                    malformed = True
+                    break
+                obj = obj_name_to_obj[object_name]
+                # Check that the type of this object agrees
+                # with what's expected given the ParameterizedOption.
+                if type_name != option.types[i].name:
+                    logging.info(f"Line {option_str} output by LLM has an "
+                                 "invalid type name.")
+                    malformed = True
+                    break
+                objs_list.append(obj)
+            if not malformed:
+                option_plan.append((option, objs_list))
+        return option_plan
+
+    def _generate_guess(self, sentence: str) -> str:
+        alphabet = [
+            'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
+            'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+            ' '
```

Comment:
wait, if the `' '` character is part of this, that means a predicate like `On` could become `a b c`, which might be interpreted as 3 separate things right? I feel like we shouldn't have spaces here.